include/helib/Matrix.h

```diff
@@ -390,7 +431,14 @@ class Tensor
   }
 
   // Matrix special
-  Tensor<T, 2>& transpose()
+  Tensor<T, 2> transpose() const
+  {
+    auto ret = this->deepCopy();
+    return ret.inPlaceTranspose();
```

Comment:
```suggestion
    ret.inPlaceTranspose();
    return ret;
```

This will enable RVO (if any)